Add Offline UX Considerations article

tomayac opened the pull request:
Changes proposed in this pull request:

- Add the article

kaycebasques commented:
@tomayac can you also create (and link to) a PR on WebFu that deletes the old content (including all translated pages) and sets up the redirect (of course we should land this first but I just want to note it before I forget)

tomayac commented:
> @tomayac can you also create (and link to) a PR on WebFu that deletes the old content (including all translated pages) and sets up the redirect (of course we should land this web.dev PR first but I just want to note the related work before I forget)

See https://github.com/google/WebFundamentals/pull/9003.

kaycebasques commented:
@tomayac are you waiting on a review from @mustafa-x?

tomayac commented:
> @tomayac are you waiting on a review from @mustafa-x?

I actually was. Else, someone else from the current web.dev team can obviously take over.

jpmedley commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+
+When designing for unstable network connections, use these:
+
+- Think how you design for the success, failure and instability of a network connection.
```
```suggestion
- Think how you design for the success, failure, and instability of a network connection.
```

jpmedley commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+- Think how you design for the success, failure and instability of a network connection.
+- Data may be expensive, so be considerate to the user.
+- For most users globally, the tech environment is almost exclusively mobile.
+- Low end devices are commonplace, with limited storage, memory and processing power and, small
```
```suggestion
- Low end devices are commonplace, with limited storage, memory and processing power, and small
```

jpmedley commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+- Try to provide offline by default if your app doesn't require much data.
+- If the app is data-heavy, educate users about how they can download for offline use.
+- Make experiences transferable between devices.
+- Utilize language, icons, imagery, typography and color to express ideas to the user collectively.
```
```suggestion
- Utilize language, icons, imagery, typography, and color to express ideas to the user collectively.
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+- Extreme weather conditions.
+- Power outages.
+- Users travelling into "dead zones" such as buildings that block their network connections.
+- Travelling on a train and going through a tunnel.
```
The difference is that being in a building is a permanent dead zone, whereas traveling through a tunnel is a temporary dead zone.

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+- Extreme weather conditions.
+- Power outages.
+- Entering into "dead zones" such as buildings that block their network connections.
+- Travelling on a train and going through a tunnel.
```
```suggestion
- Entering into temporary "dead zones" like when traveling on a train and going through a tunnel.
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+- Poor coverage of a provider.
+- Extreme weather conditions.
+- Power outages.
+- Entering into "dead zones" such as buildings that block their network connections.
```
```suggestion
- Entering into permanent "dead zones" such as buildings with walls that block network connections.
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+
+<figure class="w-figure" style="display: inline-block; max-width: 45%;">
+  <img src="emojoy-toast-message.png"
+       alt=" Emojoy, the emoji messaging app informing the user when a change in state occurs."
```
Emojoy is the name of the app.

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+
+<figure class="w-figure" style="display: inline-block; max-width: 45%;">
+  <img src="emojoy-toast-message.png"
+       alt=" Emojoy, the emoji messaging app informing the user when a change in state occurs."
```
```suggestion
       alt="The Emojoy emoji messaging app informing the user when a change in state occurs."
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+
+<figure style="display: inline-block; max-width: 45%;" class="w-figure">
+  <img src="tailpiece-offline-sidebar.png"
+       alt="Tailpiece has a navigational draw that shows what sections are ready for offline use."
```
```suggestion
       alt="Tailpiece has a visual indicator that shows what sections are ready for offline use."
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+
+<figure style="display: inline-block; max-width: 45%;" class="w-figure">
+  <img src="tailpiece-offline-sidebar.png"
+       alt="Tailpiece has a navigational draw that shows what sections are ready for offline use."
```
Replaced with "visual indicator".

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+### Clarify what is accessible offline
+
+Be clear as to the option you are providing. You may need to show a tab or setting that shows an
+"offline library", so the user can easily see what they have stored on their phone and what needs to
```
```suggestion
"offline library" or [content index](/content-indexing-api/),
so the user can easily see what they have stored on their phone and what needs to
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+Be clear as to the option you are providing. You may need to show a tab or setting that shows an
+"offline library", so the user can easily see what they have stored on their phone and what needs to
+be saved. Make sure the settings are concise and be clear where the data will be stored and who has
+access to it.
```
Absolutely! Added.

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,422 @@
+so that the experience is transferable. For example, imagine a travel app losing
+a network connection mid-way through a booking. When the connection is reestablished, the app syncs
+with the user's account allowing them to continue their booking on their desktop device. Not being
+able to transfer experiences is incredibly jarring to users.
```
```suggestion
able to transfer experiences can be incredibly jarring to users.
```

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,424 @@
+connection improves so that the experience is transferable. For example, imagine a travel app losing
+a network connection mid-way through a booking. When the connection is reestablished, the app syncs
+with the user's account and then they can continue their booking on their desktop device. Not being
+able to transfer experiences is incredibly jarring to users.
```
Weakened the statement.

tomayac commented:
Thanks, @jpmedley, I've addressed the helpful review remarks. PTAL.

kaycebasques commented:
Per our [inactive PR review policy](https://web.dev/handbook/pull-requests/#claim-an-inactive-pr) I will continue this review now

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+standard iconography, and meaningful imagery that will guide the user to complete the action or task
+rather than hinder their progress.
+
+### Use simple concise language
```
```suggestion
### Use simple, concise language
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+  </figcaption>
+</figure>
+
+### Use multiple design devices to create accessible user experiences
```
This section reinforces my previous suggestion to rephrase that section heading that is using the word `accessible` in a more general way

tomayac replied:
In this context I think "accessible" is correct, since the text talks about color deficiencies.

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+  <figcaption class="w-figcaption">
+    <b>DON'T</b>: Use color as the sole means to describe what is happening.
+  </figcaption>
+</figure>
```
Let's use a [compare component](https://web.dev/handbook/web-dev-components/#compare) here?

tomayac replied:
Applied, here and above.

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+a typical download icon or perhaps if the action involves syncing it could be a syncing icon. Some
+actions may be interpreted as saving for offline rather than demonstrating a network's status. Think
+of the action you are trying to convey rather than presenting the user with an abstract concept. For
+example, save or download data would be action based.
```
```suggestion
example, save or download data would be action-based.
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+
+<img src="download-icons-examples.png" alt="Various icon examples that convey offline">
+
+Offline can mean a number of things depending on the context, such as download, export, pin etc.
+For more inspiration checkout the
+[material design icon set](https://material.io/icons/ 'material design icon set')
```
```suggestion
Offline can mean a number of things depending on the context, such as download, export, pin, etc.
For more inspiration checkout the
[Material Design icon set](https://material.io/resources/icons/).
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+
+## Design for the next billion
+
+In many regions, low end devices are commonplace, connectivity is unreliable and, for many users,
```
```suggestion
In many regions, low-end devices are commonplace, connectivity is unreliable and, for many users,
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+
+## Conclusion
+
+Education is key to this as users are unfamiliar with these concepts. Try to create associations
```
```suggestion
Education is key to offline UX as users are unfamiliar with these concepts. Try to create associations
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+Education is key to this as users are unfamiliar with these concepts. Try to create associations
+with things that are familiar, e.g downloading for later use is the same as offlining data.
+
+When designing for unstable network connections, use these:
```
```suggestion
When designing for unstable network connections, remember these guidelines:
```

kaycebasques commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+- Think how you design for the success, failure, and instability of a network connection.
+- Data may be expensive, so be considerate to the user.
+- For most users globally, the tech environment is almost exclusively mobile.
+- Low end devices are commonplace, with limited storage, memory and processing power, and small
```
```suggestion
- Low-end devices are commonplace, with limited storage, memory and processing power, and small
```

kaycebasques commented:
@tomayac PTAL

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+---
+layout: post
+title: Offline UX Considerations
```
Absolutely.

tomayac commented on src/site/_data/paths/progressive-web-apps.json:
```diff
@@ -16,7 +16,9 @@
       "title": "Reliable",
-      "pathItems": []
+      "pathItems": [
+        "offline-ux-considerations"
```
```suggestion
        "offline-ux-design-guidelines"
```

mustafa-x commented:
Sorry for not responding sooner, I have been out dealing with a family emergency

tomayac commented on src/site/_data/paths/progressive-web-apps.json:
```diff
@@ -16,7 +16,9 @@
-      "pathItems": []
+      "pathItems": [
+        "offline-ux-considerations"
+      ]
```
Moved.

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+  <figcaption class="w-figcaption">
+    …letting them know that they can at least partially still use the app.
+  </figcaption>
+</figure>
```
Added an aside.

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+
+When a web app first loads you need to indicate to the user if it is ready for offline use. Do this
+with a
+[widget that provides brief feedback](https://material.google.com/components/snackbars-toasts.html 'widget that provides brief feedback')
```
Updated to https://material.io/components/snackbars

tomayac commented on src/site/content/en/progressive-web-apps/offline-ux-considerations/index.md:
```diff
@@ -0,0 +1,423 @@
+Again think of the language you are using to make sure it is fit for your audience. Ensure the
+messaging is given the same in all instances where it's used. The term offline is generally
+misunderstood by a non-technical audience so use action based language that your audience can relate
+to.
```
Addressed by the new aside I think.

tomayac commented:
> Sorry for not responding sooner, I have been out dealing with a family emergency

No worries, @mustafa-x, I've taken you off the reviewers list. Family is always a priority. @kaycebasques, I have addressed your remarks. PTAL.

kaycebasques commented on src/site/content/en/reliable/offline-ux-design-guidelines/index.md:
```diff
@@ -0,0 +1,441 @@
+To prevent misunderstandings, express different states to the user in multiple ways, for example
+with color, labels, and UI components.
+
+{% Compare 'worse' %}
+  <figure style="display: inline-block; max-width: 45%;" class="w-figure">
+    <img src="accessibility_color8_dont.png" alt="A bad example only using color.">
+    <figcaption class="w-figcaption">
+      Use color as the sole means to describe what is happening.
+    </figcaption>
+  </figure>
+{% endCompare %}
+
+{% Compare 'better' %}
+  <figure style="display: inline-block; max-width: 45%;" class="w-figure">
+    <img src="accessibility_color7_do.png" alt="A good example that uses color and text to show an error.">
+    <figcaption class="w-figcaption">
+      Use a mixture of design elements to convey meaning
+    </figcaption>
+  </figure>
+{% endCompare %}
```
@tomayac I won't block you on this but you might enjoy using columns to get these to display side-by-side. See https://web.dev/css-marker-pseudo-element/ for an example:

![Uploading image.png…]()

kaycebasques reviewed (approved):
LGTM, thanks @tomayac

kaycebasques commented:
@tomayac can you create a PR to redirect the old content?

tomayac commented:
@kaycebasques Tagged you on the WebFu PR, and opened https://github.com/GoogleChrome/web.dev/pull/4142 as a followup to your suggestion in https://github.com/GoogleChrome/web.dev/pull/3863#discussion_r513737612.
